src/app/core/pipes/format-datetime.pipe.ts

```diff
@@ -20,19 +23,44 @@ export class FormatDateTimePipe implements PipeTransform {
 
   constructor(
     private store$: Store<AppState>,
+    private actions$: Actions,
     private cdr: ChangeDetectorRef,
     @Inject(WINDOW) private window: Window,
   ) {
     this.store$.select(selectTimezone).pipe(untilDestroyed(this)).subscribe((timezone) => {
       this.timezone = timezone;
       this.cdr.markForCheck();
+      this.checkFormatsFromLocalStorage();
+    });
+
+    this.actions$
```

Comment:
I'd like to mention that checking for changes in preferences from the store allows us to avoid new entity injection. However, I don't know if that has a noticeable difference.